fcrepo4/fcrepo4
# Implement kernel events

**JIRA Ticket**: [FCREPO-3249](https://jira.lyrasis.org/browse/FCREPO-3249) & [FCREPO-3250](https://jira.lyrasis.org/browse/FCREPO-3250)

# What does this Pull Request do?

ResourceOperations are turned into Events that are emitted to the JMS code when a transaction is successfully completed.

# How should this be tested?

1. Start Fedora
1. [Start an event listener](https://wiki.lyrasis.org/display/FEDORA51/How+to+Inspect+Event+Messages+Generated+by+Fedora)
1. Perform a mutating action on a resource
1. See a JMS event logged by the listener

# Additional Notes:

* Events are only emitted if a transaction is entirely successful
* I don't think that `FedoraResource` is correctly loading all of a resources types at this point

# Interested parties

@fcrepo4/committers
